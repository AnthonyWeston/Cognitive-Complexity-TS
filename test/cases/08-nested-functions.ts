function f() {
    function ff() {

    }
}

function g() {
    const gg = () => {};
}

function h() {
    function hh() {
        while (true);
    }
}

// depth 0
function i() {
    // depth 0
    const ii = () => {
        // depth 1
        while (true); // +1 +1 for depth
    };
}
