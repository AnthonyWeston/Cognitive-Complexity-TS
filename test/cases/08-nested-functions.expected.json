{
    "08-nested-functions.ts": {
        "score": 2,
        "inner": [
            {
                "name": "f",
                "score": 0,
                "line": 1,
                "column": 1,
                "inner": [
                    {
                        "name": "ff",
                        "score": 0,
                        "line": 2,
                        "column": 4
                    }
                ]
            },
            {
                "name": "g",
                "score": 0,
                "line": 5,
                "column": 1,
                "inner": [
                    {
                        "name": "gg",
                        "score": 0,
                        "line": 8,
                        "column": 4
                    }
                ]
            },
            {
                "name": "h",
                "score": 1,
                "line": 11,
                "column": 1,
                "inner": [
                    {
                        "name": "hh",
                        "score": 1,
                        "line": 12,
                        "column": 4
                    }
                ]
            },
            {
                "name": "i",
                "score": 2,
                "line": 19,
                "column": 1,
                "inner": [
                    {
                        "name": "ii",
                        "score": 1,
                        "line": 20,
                        "column": 4
                    }
                ]
            }
        ]
    }
}